### PR TITLE
update min requirement for py3.5, py3.6 on win32

### LINKED ISSUE #37: the minimum version for windows should be 1.14
before numpy 1.14, on windows, if an extension included npymath, there would be [compiler incompatibility with the windows numpy wheel](https://github.com/numpy/numpy/issues/9958).

This was fixed in 1.14. For `win32`, that should be the minimum version.

### PR DESCRIPTION
**Pull request by fgregg**
closes #37

(I can rebase this into a single clean commit, if this is the way you want to go)

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -38,13 +38,17 @@ install_requires =
     # (first version with universal2 wheels available)
     numpy==1.21.0; python_version=='3.8' and platform_machine=='arm64' and platform_system=='Darwin'
     numpy==1.21.0; python_version=='3.9' and platform_machine=='arm64' and platform_system=='Darwin'
+    
+    # If a C extension uses npymath, then Windows needs a bugfix in numpy>=1.14.1
```
Comment (bashtage) [inline setup.cfg:42]: Why do the comments and the pins not match?  Comments says >= 1.14.1 but pin is for 1.14.0. Should pin be 1.14.5, which is the last release of 1.14 which has the same C interface as 1.14.0?

Comment (fgregg) [reply setup.cfg:42]: as for 1.14.5 as the preferred choice, i’ll defer to others on that

Comment (rgommers) [reply setup.cfg]: I had a look at this for another issue recently, and it should indeed be fine to compile against the latest bugfix release of a minor release (so `1.14.5`) and then use it with all `1.14.x` releases.

Comment (fgregg) [reply setup.cfg]: fixed comment and bumped to 1.14.5

File: setup.cfg [hunk 2]
```diff
@@ -38,13 +38,17 @@ install_requires =
...
 
     # Python 3.8 on s390x requires at least 1.17.5, see gh-29
     numpy==1.17.5; python_version=='3.8' and platform_machine=='s390x' and platform_python_implementation != 'PyPy'
 
     # default numpy requirements
-    numpy==1.13.3; python_version=='3.5' and platform_machine!='aarch64' and platform_system!='AIX'
-    numpy==1.13.3; python_version=='3.6' and platform_machine!='aarch64' and platform_system!='AIX' and platform_python_implementation != 'PyPy'
+    numpy==1.13.3; python_version=='3.5' and platform_machine!='aarch64' and platform_system!='AIX' and platform_system!='Windows'
+    numpy==1.13.3; python_version=='3.6' and platform_machine!='aarch64' and platform_system!='AIX' and platform_system!='Windows' and platform_python_implementation != 'PyPy'
     numpy==1.14.5; python_version=='3.7' and platform_machine!='aarch64' and platform_system!='AIX' and platform_python_implementation != 'PyPy'
```
[the remark this answers is not in the document]
Comment (fgregg) [reply setup.cfg:52]: bumped to 1.14.5

**Review by rgommers — COMMENTED** [review]
No other comments on https://github.com/scipy/oldest-supported-numpy/issues/37#issuecomment-999748617, just one thumbs up. So I think I'll reject this PR. The changes are good for packages that build against `npymath` with MSVC 2015, but it's a regression for others (it completely breaks using `1.13.x`). So I think the best resolution here is to take over the code comment here, and ask projects who indeed need this change to copy the whole contents of `setup.cfg` including this change into their `pyproject.toml`, rather than relying on `oldest-supported-numpy`.

**Comment by fgregg** [comment]
thanks for considering. another option for guidance could be to do a pyproject.toml like this https://github.com/dedupeio/pyhacrf/blob/97beade4796e232c4f8b0a2c5c2e25764bd49201/pyproject.toml#L5

**Comment by rgommers** [comment]
> thanks for considering. another option for guidance could be to do a pyproject.toml like this https://github.com/dedupeio/pyhacrf/blob/97beade4796e232c4f8b0a2c5c2e25764bd49201/pyproject.toml#L5

Great point, that's even better, much easier to read.

**Comment by rgommers** [comment]
Maybe that's even worth adding to the Q&A section of the README.

**Comment by fgregg** [comment]
this is a touch more future proof:

https://github.com/dedupeio/pyhacrf/blob/c6f7265404938dbfc8ea94889a79319b4889eeb7/pyproject.toml#L5

**Comment by rgommers** [comment]
I'll close this. Thanks again for working through this @fgregg

**Pull request closed without merging**